src/processor/snapshot/mod.rs

```diff
@@ -1,74 +1,49 @@
-use std::{collections::HashMap, fmt, fs, path::PathBuf, str::FromStr};
+use std::{fs, path::PathBuf, str::FromStr};
 
 mod bytecode;
+mod database;
 mod types;
 
 use async_trait::async_trait;
 use blake2::{Blake2s256, Digest};
 use ethers::types::{Address, H256, U256, U64};
 use eyre::Result;
-use indexmap::IndexSet;
 use state_reconstruct_fetcher::{
     constants::{ethereum, storage},
     types::CommitBlockInfoV1,
 };
 use tokio::sync::mpsc;
 
-use self::types::{SnapshotFactoryDependency, SnapshotStorageLog, StorageKey, StorageValue};
+use self::{
+    database::{DatabaseError, SnapshotDB},
```

Comment:
```suggestion
    database::SnapshotDB,
```

I don't think this is used anywhere, should it be?

Reply:
Depends on incarnation of code. There is a specific error for "no such key" case, but apparently the need for specific type dropped out. :slightly_smiling_face: